player.py

```diff
@@ -6,7 +6,7 @@ def __init__(self,piece):
         
     
     def get_name(self):
-        self = input("What is your name:  ")
+        self = input("What is your name:  ")  # bad idea to use self as the variable name
```

Comment:
self is the Player object itself.  In this case, it didn't seem to cause a problem, but it is definitely confusing as to why it didn't.  Use a different variable name, or just return the input statement.